test/sdp_interop.js

```diff
@@ -744,3 +744,218 @@ a=rtcp-mux\r\n"
+a=setup:active\r\n\
+a=mid:audio\r\n\
+a=extmap:3 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time\r\n\
+a=recvonly\r\n\
+a=rtcp-mux\r\n\
+a=rtpmap:96 opus/48000/2\r\n\
+a=fmtp:96 minptime=10; useinbandfec=1\r\n\
+a=rtpmap:0 PCMU/8000\r\n\
+a=maxptime:60\r\n\
+m=video 9 RTP/SAVPF 99\r\n\
+c=IN IP4 0.0.0.0\r\n\
+a=rtcp:9 IN IP4 0.0.0.0\r\n\
+a=ice-ufrag:D/SIZC9xg+MActHq\r\n\
+a=ice-pwd:0AJ690SNxw0UHVNmIZDEBHQy\r\n\
+a=fingerprint:sha-256 D2:7A:84:32:AE:19:4A:C0:96:33:36:61:81:F2:BD:2C:57:9E:79:EE:56:5B:1D:19:37:E4:AC:ED:70:C8:79:72\r\n\
+a=setup:active\r\n\
+a=mid:video\r\n\
+a=extmap:3 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time\r\n\
+a=recvonly\r\n\
+a=rtcp-mux\r\n\
+a=rtpmap:99 VP8/90000\r\n\
+a=rtcp-fb:99 ccm fir\r\n\
+a=rtcp-fb:99 nack\r\n\
+a=rtcp-fb:99 nack pli\r\n\
+a=rtcp-fb:99 goog-remb"
+
+  /*jshint multistr: true */
+  var expectedUnifiedPlan =
+    "v=0\r\n\
+o=- 590499500270831419 2 IN IP4 127.0.0.1\r\n\
+s=-\r\n\
+t=0 0\r\n\
+a=msid-semantic: WMS *\r\n\
+a=group:BUNDLE audio0 video0 video1\r\n\
+m=audio 1 RTP/SAVPF 96 0\r\n\
+a=rtpmap:96 opus/48000/2\r\n\
+a=rtcp:9 IN IP4 0.0.0.0\r\n\
+a=extmap:3 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time\r\n\
+a=mid:audio0\r\n\
+a=recvonly\r\n\
+a=ice-ufrag:D/SIZC9xg+MActHq\r\n\
+a=ice-pwd:0AJ690SNxw0UHVNmIZDEBHQy\r\n\
+a=fingerprint:sha-256 D2:7A:84:32:AE:19:4A:C0:96:33:36:61:81:F2:BD:2C:57:9E:79:EE:56:5B:1D:19:37:E4:AC:ED:70:C8:79:72\r\n\
+a=rtcp-mux\r\n\
+m=video 1 RTP/SAVPF 99\r\n\
+b=AS:2000\r\n\
+a=rtpmap:99 VP8/90000\r\n\
+a=rtcp:9 IN IP4 0.0.0.0\r\n\
+a=rtcp-fb:99 nack\r\n\
+a=rtcp-fb:99 nack pli\r\n\
+a=rtcp-fb:99 goog-remb\r\n\
+a=rtcp-fb:99 ccm fir\r\n\
+a=extmap:3 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time\r\n\
+a=mid:video0\r\n\
+a=recvonly\r\n\
+a=ice-ufrag:D/SIZC9xg+MActHq\r\n\
+a=ice-pwd:0AJ690SNxw0UHVNmIZDEBHQy\r\n\
+a=fingerprint:sha-256 D2:7A:84:32:AE:19:4A:C0:96:33:36:61:81:F2:BD:2C:57:9E:79:EE:56:5B:1D:19:37:E4:AC:ED:70:C8:79:72\r\n\
+a=rtcp-mux\r\n\
+m=video 1 RTP/SAVPF 101\r\n\
+b=AS:2000\r\n\
+a=rtpmap:101 VP8/90000\r\n\
+a=rtcp:9 IN IP4 0.0.0.0\r\n\
+a=rtcp-fb:101 nack\r\n\
+a=rtcp-fb:101 nack pli\r\n\
+a=rtcp-fb:101 goog-remb\r\n\
+a=rtcp-fb:101 ccm fir\r\n\
+a=extmap:3 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time\r\n\
+a=mid:video1\r\n\
+a=recvonly\r\n\
+a=ice-ufrag:D/SIZC9xg+MActHq\r\n\
+a=ice-pwd:0AJ690SNxw0UHVNmIZDEBHQy\r\n\
+a=fingerprint:sha-256 D2:7A:84:32:AE:19:4A:C0:96:33:36:61:81:F2:BD:2C:57:9E:79:EE:56:5B:1D:19:37:E4:AC:ED:70:C8:79:72\r\n\
+a=rtcp-mux\r\n"
+
+  var interop = new Interop();
+
+  var offer = new RTCSessionDescription({
+    type: 'offer',
+    sdp: originUnifiedPlan
+  });
+
+  var planBDesc = interop.toPlanB(offer);
+  assert.equal(planBDesc.sdp, expectedPlanB,
+    "Not expected Plan B output")
+
+  var answer = new RTCSessionDescription({
+    type: 'answer',
+    sdp: originPlanB
+  });
+
+  var unifiedPlanDesc = interop.toUnifiedPlan(answer);
```

Comment:
In the vanilla sdp-interop, this is the step that fails with an exception from line https://github.com/jitsi/sdp-interop/blob/81f4186d905bda25bcda21f3480f76e144111932/lib/interop.js#L594 complaining about "TypeError: Cannot read property 'media' of undefined" because cached is undefined. In jitsi-meet (where FF is always the answerer), prior to calling toUnifiedPlan(bAnswer), we call toPlanB(uAnswer) which populates the cache. The issue here is that simply using the inverted the "cached" object won't solve the general problem which is that the library assumes that FF is always the answerer https://github.com/jitsi/sdp-interop/issues/4.